lib/swig/GeNNGroups.py

```diff
@@ -0,0 +1,477 @@
+import pygenn as pg
+
+class Variable(object):
+
+    """Class holding information about GeNN variables"""
+
+    def __init__(self, variableName, variableType, values=None):
+        """Init Variable
+
+        Args:
+        variableName -- string name of the variable
+        variableType -- string type of the variable
+
+        Keyword args:
+        values       -- iterable or sigle value.
+        """
+        self.name = variableName
+        self.type = variableType
+        self.view = None
+        self.needsAllocation = False
+        if values is not None:
+            self.setValues( values )
+
+    def setInitVar( self, initVarSnippet, paramSpace ):
+        """Set variable initialization using InitVarSnippet
+
+        Args:
+        initVarSnippet -- type as string or instance of a class derived from InitVarSnippet.Custom
+        paramSpace     -- dict mapping parameter names to their values for InitVarSnippet
+        """
+        ivsInst, ivsType = ModelPreprocessor.isModelValid( initVarSnippet, pg.InitVarSnippet )
+        params = ModelPreprocessor.parameterSpaceToParamValues( ivsInst, paramSpace )
+        initFct = getattr( pg, 'initVar_' + ivsType )
+        self.initVal = initFct( params )
+
+    def setValues( self, values, initVar=None ):
+        """Set Variable's values
+
+        Args:
+        values -- iterable or single value or parameter space is initVar is specified
+
+        Keyword args:
+        initVar -- type as string or instance of a class derived from InitVarSnippet.Custom
+        """
+        self.initRequired = False
+        if initVar is not None:
+            setRandomInit( initVar, values )
```

Comment:
Where is this function located and what does it do? Are InitVarSnippets implemented or should that be added to the Todo?

Reply:
`InitVarSnippet`s were the last thing I implemented (actually just yesterday) and are not thoroughly tested, but they hopefully work. `self.initRequired`'s purpose is to indicate that a list of values was provided and if it is true, these values will be pushed to device when the model is loaded.

Reply:
Cool that they're implemented! What I meant though is where is ``setRandomInit``? I can't find it in the module...

Reply:
oops that's a typo. It must read `self.setInitVar`.